src/server/raft_client.rs

```diff
@@ -1213,8 +1300,355 @@ where
+                max_delays,
+                shutdown_rx,
+            )
+            .await;
+        });
+    }
+
+    /// Stop inspection task for a specific store
+    fn stop_store_inspection(&self, store_id: u64) {
+        info!("Stopping health check inspection for store"; "store_id" => store_id);
+
+        let shutdown_tx = {
+            let mut handles = self.task_handles.lock().unwrap();
+            handles.remove(&store_id)
+        };
+
+        if let Some(tx) = shutdown_tx {
+            let _ = tx.send(());
+        }
+    }
+
+    /// Main inspection loop for a single store
+    async fn store_inspection_loop(
+        self_store_id: u64,
+        store_id: u64,
+        pool: Arc<Mutex<ConnectionPool>>,
+        inspect_network_interval: Arc<Mutex<Duration>>,
+        max_delays: Arc<Mutex<HashMap<u64, f64>>>,
+        mut shutdown_rx: oneshot::Receiver<()>,
+    ) {
+        let mut last_check_time = Instant::now();
+
+        loop {
+            // Check if we should shutdown
+            if let Ok(Some(())) = shutdown_rx.try_recv() {
+                info!("Shutting down health check inspection for store"; "store_id" => store_id);
+                // Reset max delay for this store
+                {
+                    let mut max_delays = max_delays.lock().unwrap();
+                    max_delays.remove(&store_id);
+                }
+                break;
+            }
+
+            // Read the current interval
+            let interval = {
+                let guard = inspect_network_interval.lock().unwrap();
+                *guard
+            };
+
+            // Wait for the specified interval
+            if let Err(e) = GLOBAL_TIMER_HANDLE
+                .delay(last_check_time + interval)
+                .compat()
+                .await
+            {
+                error!("failed to delay in network inspection"; "store_id" => store_id, "error" => ?e);
+                continue;
+            }
+
+            last_check_time = Instant::now();
+
+            // Get connections for this specific store
+            let store_connections: Vec<(usize, Arc<Queue>, Option<Channel>)> = {
+                let pool_guard = pool.lock().unwrap();
+                pool_guard
+                    .connections
+                    .iter()
+                    .filter_map(|((sid, conn_id), conn_info)| {
+                        if *sid == store_id {
+                            Some((*conn_id, conn_info.queue.clone(), conn_info.channel.clone()))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect()
+            };
+
+            // If no connections for this store, continue
+            if store_connections.is_empty() {
+                continue;
+            }
+
+            // Perform health check for each connection of this store
+            for (conn_id, queue, channel_opt) in store_connections {
```

Comment:
Do we need to loop all connections?

Reply:
Good catch! We can exit after `perform_health_check`